ui-charts/src/spaceTimeChart/utils/canvas.ts

```diff
@@ -323,24 +323,22 @@ export function drawPathExtremity(
 }
 
 /**
- *
  * @param minT number timestamp
  * @param maxT number timestamp
  * @param timeRanges time frames (24h, 12h, 6h, …)
  * @param gridlinesLevels width of the lines for each time frame
  * @param formatter function to format de values inside the output object
- * @returns Record<number, number>
  * Keys are times in ms
  * Values are the highest level on each time
  */
-export function computeVisibleTimeMarkers<T>(
+export function computeVisibleTimeMarkers<T extends object = { level: number }>(
   minT: number,
   maxT: number,
   timeRanges: number[],
   gridlinesLevels: number[],
-  formatter: (level: number, i: number) => T = identity
-) {
-  const result: Record<number, T> = {};
+  formatter: (level: number, i: number) => T = (level: number) => ({ level }) as T
+): (T & { time: number })[] {
+  const result: Record<number, T & { time: number }> = {};
```

Comment:
nit: you can directly create an array
```suggestion
  const result: (T & { time: number })[] = [];
```

Reply:
No, I can't, because I use keys overriding in the object, to only preserve the highest level for each time, to avoid specifying seconds where we also specify hours, for instance.